lib/image-stream/utils.js

```diff
@@ -97,3 +97,28 @@ exports.extractStream = (stream) => {
     });
   });
 };
+
+/**
+ * @summary Get image extension from archive name
+ * @function
+ * @public
+ *
+ * @param {String} archiveName - archive name
+ * @returns {String} - image extension
+ *
+ * @example
+ * const extension = utils.getImageExtension('quirky-pi2-sd-8gb-xerus-8.1.img.xz');
+ * console.log(extension);
+ *
+ * > '.img'
+ */
+exports.getImageExtension = (archiveName) => {
```

Comment:
Can we somehow re-use the logic from https://github.com/resin-io/etcher/blob/master/lib/shared/supported-formats.js#L109? E.g: maybe this function can use the extension retrieval logic from `supported-formats.js`, and then `isSupportedImage()` can re-use this function.